### PR TITLE
nfs: check if /sbin/mount.nfs* is present in CanSupport()

### PR DESCRIPTION
**Pull request by rootfs**
Check if nfs mount is installed and help diagnose pod creation problems.

@kubernetes/rh-storage

### CODE REVIEW

File: pkg/volume/nfs/nfs.go [hunk 1]
```diff
@@ -66,9 +67,21 @@ func (plugin *nfsPlugin) Name() string {
...
 
+func findMount() bool {
+	exe := exec.New()
+	cmd1 := exe.Command("/bin/ls", "/sbin/mount.nfs")
+	_, err1 := cmd1.CombinedOutput()
+	cmd2 := exe.Command("/bin/ls", "/sbin/mount.nfs4")
+	_, err2 := cmd2.CombinedOutput()
+	return (err1 == nil || err2 == nil)
+}
+
```
Comment (markturansky) [inline pkg/volume/nfs/nfs.go]: Do you think we should make conditional compilation files so that a linux check can differ from other platforms?

Comment (rootfs) [inline pkg/volume/nfs/nfs.go]: I only tested fedora/rhel and ubuntu. This two-pass check is for platforms I haven't tested.

Comment (markturansky) [inline pkg/volume/nfs/nfs.go]: On OSX, for example, it's "/sbin/mount_nfs".

File: pkg/volume/nfs/nfs_test.go [hunk 2]
```diff
@@ -38,12 +38,16 @@ func TestCanSupport(t *testing.T) {
 	if plug.Name() != "kubernetes.io/nfs" {
 		t.Errorf("Wrong name: %s", plug.Name())
 	}
-	if !plug.CanSupport(&volume.Spec{Volume: &api.Volume{VolumeSource: api.VolumeSource{NFS: &api.NFSVolumeSource{}}}}) {
+	foundMount := findMount()
+	if plug.CanSupport(&volume.Spec{Volume: &api.Volume{VolumeSource: api.VolumeSource{NFS: &api.NFSVolumeSource{}}}}) != foundMount {
```
Comment (markturansky) [inline pkg/volume/nfs/nfs_test.go]: I think this will be false on OSX.  The 2 paths used in findMount do not exist, so return false.

Comment (rootfs) [inline pkg/volume/nfs/nfs_test.go]: I'll get a mac next time :)
osx (darwin) is added now.

File: pkg/volume/nfs/nfs.go (hunk 1 above)
Comment (markturansky) [inline pkg/volume/nfs/nfs.go]: Looks good with the switch.

File: pkg/volume/nfs/nfs.go [hunk 3]
```diff
@@ -66,9 +68,29 @@ func (plugin *nfsPlugin) Name() string {
 	return nfsPluginName
 }
 
+func findMount() bool {
```
Comment (markturansky) [inline pkg/volume/nfs/nfs.go]: is `isSupported` or `hasOSSupport` a better name?  What exactly are you testing?

Comment (rootfs) [inline pkg/volume/nfs/nfs.go]: name updated

**Comment by markturansky** [comment]
LGTM.

@kubernetes/rh-storage 2nd opinion?  Particularly around the syscalls being made.

**Comment by childsb** [comment]
this is great, will be nice when storage errors are more verbose too! 

+1

**Comment by saad-ali** [comment]
LGTM

**Comment by markturansky** [comment]
@k8s-bot test this please

**Comment by markturansky** [comment]
all green and lgtm from saad.  adding ok-to-merge label.

**Pull request merged**